app/models/course/condition.rb

```diff
@@ -5,4 +5,9 @@ class Course::Condition < ActiveRecord::Base
   belongs_to :conditional, polymorphic: true
 
   delegate :satisfied_by?, to: :actable
+
+  ALL_CONDITIONS = [
+    Course::Condition::Achievement.class_name.downcase,
```

Comment:
It's not documented, and you're breaking abstraction barriers. `url_for` is a view/controller helper concept. You've conflated with model which you are defining with the view and controller.